rte/kernels-openacc/mo_rte_solver_kernels.F90

```diff
@@ -1197,5 +1198,310 @@ subroutine apply_BC_0(ncol, nlay, ngpt, top_at_1, flux_dn) bind (C, name="apply_
       end do
     end if
   end subroutine apply_BC_0
+  ! ---------------------------------------------------------------
+  !
+  ! Tang
+  !
+  ! ---------------------------------------------------------------
+  subroutine lw_solver_1rescl(ncol, nlay, ngpt, top_at_1, D, weight,                             &
```

Comment:
Consistent a comment below this seems like a lot of repeated code. Could `lw_solver_1rescl()` be formulated to call `lw_solver_1str()` and then adjust the fluxes in a second step?

Reply:
The radiance fro the next level depends on the radiance on the previous level. So the adjustment to a radiance at given level has to be propagated to the followings levels. It seems to be simpler doing that way despite a lot of repeated code. As I suggested some  repeated code can be moved to lw_transport_1scl if representation for the adjustmant factor as 
adjustmentFactor = A_up(dn) radn_dn(up) + B_up(dn) source_dn(up)+C_dn(up)
is used